webpack/assets/javascripts/react_app/components/HostsIndex/BulkActions/buildHosts/actions.js

```diff
@@ -0,0 +1,26 @@
+import { translate as __ } from '../../../../common/I18n';
+import { API_OPERATIONS, APIActions } from '../../../../redux/API';
+import { foremanUrl } from '../../../../common/helpers';
+
+export const bulkBuildHosts = (params, handleSuccess, handleError) => {
+  const errorToast = ({ message }) => message;
+  const url = foremanUrl(`/api/v2/hosts/bulk/build`);
+  return APIActions.put({
+    type: API_OPERATIONS.PUT,
+    key: 'HOST_BUILD_KEY',
+    url,
+    successToast: () => {
+      if (params.rebuild_configuration)
+        return __('Rebuilt the configuration for hosts.');
+      return __('Built hosts.');
+    },
+    handleSuccess: response => {
+      if (handleSuccess) handleSuccess(response);
+    },
```

Comment:
Any reason this won't work?

```suggestion
    handleSuccess,
```

APIRequest.js already handles it for us if you don't pass `handleSuccess`: https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/redux/API/APIRequest.js#L20

Reply:
updated